src/ClientHandle.cpp

```diff
@@ -1335,10 +1335,10 @@ void cClientHandle::HandleBlockDigFinished(int a_BlockX, int a_BlockY, int a_Blo
 		}
 	}
 
-	if (!m_Player->IsGameModeCreative() && !cBlockInfo::IsOneHitDig(a_OldBlock))
+	if (!m_Player->IsGameModeCreative() && !m_Player->CanInstantlyMine(a_OldBlock))
 	{
-		// Fix for very fast tools.
-		m_BreakProgress += m_Player->GetPlayerRelativeBlockHardness(a_OldBlock);
+		m_BreakProgress += m_Player->GetMiningProgressPerTick(a_OldBlock);
+		// Check for very fast tools. Maybe instead of FASTBREAK_PERCENTAGE we should check we are within x multiplied by the progress per tick
 		if (m_BreakProgress < FASTBREAK_PERCENTAGE)
```

Comment:
Any opinions on this? I think checking for x ticks missed feels more natural. Then it is coupled to how fast their mining tool is instead of just an arbitrary leeway

Reply:
That sounds like a good idea. You'll need to update `FASTBREAK_PERCENTAGE` either way:
https://github.com/cuberite/cuberite/blob/c158569af3f8303c1c2472490cfd2f5a1f599d35/src/Root.cpp#L138-L139

Reply:
Ah it's off completely, I'm not sure I want to turn it on because some of our block hardness values and CanHarvest etc seem to be wrong and this would cause people to be randomly kicked.